### DISCUSSION TITLE
Attach new file from existing attachment

### DISCUSSION THREAD

**Discussion by Faq** [shrinerb/shrine#624] [Q&A] [answered]
Hello,

Rails app.
I have set local storage _store_ and _cache_ folder and uploaded file stored in _store_. There is need to Attach copy of this uploaded file (same name and content, different id) to custom email form.
So tried to do something like:

In model is ` include Uploader::Attachment(:file)`
From form is called method with:
new(file: `somefile`)

Existing uploaded file when calling `somefile`, returns:
`<Uploader::UploadedFile storage=:store id="c2543a5f6b22b03a28987a1dda618951.docx" metadata={"size"=>121454, "filename"=>"example_name.docx", "mime_type"=>nil}>`

`new(file: somefile)`, returns:
```
#<Attachment::Letter::Document:0x000000010a216550
 id: nil,
 type: "Attachment::Letter::Document",
 file_data: {"id"=>"ca82057757da17666f11ad653cef8510.docx", "storage"=>"cache", "metadata"=>{"size"=>121454, "filename"=>"example_name.docx", "mime_type"=>nil}},
```

After form is submitted, it is stored in cache not as store.

What is preferred way to do this?

Tried but no luck, by using attacher:
` Uploader::Attacher.attach(somefile, storage: :store)`

**Comment by benkoshy** [comment, marked as answer]
I'm having trouble understanding your question?

Is it possible to post an example of what you're trying to do? https://github.com/shrinerb/shrine/blob/master/SELF_CONTAINED_EXAMPLE.md#option-2-manually-set-up-the-self-contained-example

**Reply by Faq** [reply]
Solution was found. With form should be passed `filename.attachment.file` not `filename.attachment.file_data`
